src/app/features/tech-record/components/tech-record-change-visibility/tech-record-change-visibility.component.ts

```diff
@@ -0,0 +1,95 @@
+import { Location } from '@angular/common';
+import { Component, OnInit } from '@angular/core';
+import { Validators } from '@angular/forms';
+import { ActivatedRoute, Router } from '@angular/router';
+import { GlobalErrorService } from '@core/components/global-error/global-error.service';
+import { CustomFormControl, CustomFormGroup, FormNodeTypes } from '@forms/services/dynamic-form.types';
+import { TechRecordModel, VehicleTechRecordModel } from '@models/vehicle-tech-record.model';
+import { Actions, ofType } from '@ngrx/effects';
+import { RouterReducerState } from '@ngrx/router-store';
+import { Store } from '@ngrx/store';
+import { TechnicalRecordService } from '@services/technical-record/technical-record.service';
+import { selectRouteNestedParams } from '@store/router/selectors/router.selectors';
+import { updateEditingTechRecord, updateTechRecords, updateTechRecordsSuccess } from '@store/technical-records';
+import { TechnicalRecordServiceState } from '@store/technical-records/reducers/technical-record-service.reducer';
+import cloneDeep from 'lodash.clonedeep';
+import { filter, mergeMap, Observable, take } from 'rxjs';
+
+@Component({
+  selector: 'app-tech-record-change-visibility',
+  templateUrl: './tech-record-change-visibility.component.html',
+  styleUrls: ['./tech-record-change-visibility.component.scss']
+})
+export class TechRecordChangeVisibilityComponent implements OnInit {
+  vehicleTechRecord$: Observable<VehicleTechRecordModel | undefined>;
+
+  form: CustomFormGroup;
+
+  constructor(
```

Comment:
Component needs some unit test coverage

Reply:
To prioritize the routing refactoring, I will add unit tests after we resolve that this morning.